types/node/node-tests.ts

```diff
@@ -52,12 +52,12 @@ assert['fail'](true, true, 'works like a charm');
 
 {
     const a = 'test' as any;
-    assert.strictEqual(a, 'test');
+    assert.strictEqual<string>(a, 'test');
```

Comment:
Seems this change causes that asserts like this require users to specify the type.
I don't think this is helpful.

Reply:
i don't see why the user should be allowed to specify values with non-matching types without having to explicitly specify that they're intending to do so (either via type casting or specifying generics). it seems like the cost of the amount of runtime errors that could easily be picked up at compiletime (like my [example above](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50274#issuecomment-753414752)) outweighs the slight convenience for edge cases where you'd actually want non-matching types.

that said, your example could be fixed by doing something like this, but i question whether doing so is a good idea
```ts
function strictEqual<T extends U, U>(actual: U, expected: T, message?: string | Error): asserts actual is T
```

Reply:
It's not about wanting non matching types. `any` and `string` match similar as `string | number` matches `string`.
But not the assert is no longer asserting these types correct.

Using `function strictEqual<T extends U, U>(actual: U, expected: T, message?: string | Error): asserts actual is T` sounds better as it solves at least the cases with `any` and union types which are quite common in my opinion.

Reply:
done